steamcmd_servers/arma/arma3/egg-arma3.json

```diff
@@ -1,108 +1,149 @@
             "container": "debian:buster-slim",
             "entrypoint": "bash"
         }
     },
     "variables": [
         {
-            "name": "Game App ID",
-            "description": "Steam CMD App ID.",
-            "env_variable": "SRCDS_APPID",
+            "name": "Arma 3 Dedicated Server App ID",
+            "description": "Used for installation and updates.",
+            "env_variable": "STEAMCMD_APPID",
             "default_value": "233780",
-            "user_viewable": 1,
-            "user_editable": 0,
-            "rules": "required|alpha_dash|between:1,100"
+            "user_viewable": false,
+            "user_editable": false,
+            "rules": "required|numeric"
         },
         {
-            "name": "Config",
-            "description": "Server config",
-            "env_variable": "CONFIG",
-            "default_value": "server.cfg",
-            "user_viewable": 1,
-            "user_editable": 1,
-            "rules": "string|nullable"
+            "name": "Steam Username",
+            "description": "Used for installation and updates. Account does not need to own Arma 3, but must be a valid Steam account (anonymous login cannot be used). For security reasons it is recommended that you create a new Steam account just for your dedicated servers. *This info is not visible to the server owner.*",
+            "env_variable": "STEAM_USER",
+            "default_value": "your_steam_username",
+            "user_viewable": false,
+            "user_editable": false,
+            "rules": "required|string"
+        },
+        {
+            "name": "Steam Password",
+            "description": "Used for installation and updates. Steam Guard must be TURNED OFF for install to complete successfully. *This info is not visible to the server owner.*",
+            "env_variable": "STEAM_PASS",
+            "default_value": "your_steam_password",
+            "user_viewable": false,
+            "user_editable": false,
+            "rules": "required|string"
         },
         {
-            "name": "Custom Command Line Option.",
-            "description": "Put -autoinit or related things in here :)",
-            "env_variable": "CUSTOM",
+            "name": "Server Binary",
+            "description": "Server binary to call on server start. (32-bit: arma3server | 64-bit: arma3server_x64)",
+            "env_variable": "SERVER_BINARY",
+            "default_value": "arma3server_x64",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string"
```

Comment:
You can make this only accept working values with `in:value1,value2`

Reply:
Oh, that's actually really cool; I did not know that was possible. I could definitely change it to `in:arma3server,arma3server_x64,arma3serverprofiling,arma3serverprofiling_x64` if you would like. However, in the end, I'm okay with leaving it open ended because other builds or future beta builds may have different binary names. Ensuring the list is always up to date could be troublesome, and I believe leaving this variable open-ended allows more freedom for the end user.

Reply:
On the topic though... Does Ptero support the `regex:pattern` rule? I couldn't get it to work for the basic.cfg and server.cfg fields. I just wanted to make sure the string provided ended in `.cfg`, but I couldn't get the `regex:pattern` or `file` rules to work (got 500 internal server errors).

Reply:
> On the topic though... Does Ptero support the `regex:pattern` rule? I couldn't get it to work for the basic.cfg and server.cfg fields. I just wanted to make sure the string provided ended in `.cfg`, but I couldn't get the `regex:pattern` or `file` rules to work (got 500 internal server errors).

Pterodactyl follows the laravel validation rules, regex works just fine, remember to separate your rule with forward slash ex. `required|regex:/^[a-zA-Z0-9]*$/|max:64`